node/src/components/event_stream_server.rs

```diff
@@ -58,13 +61,41 @@ pub(crate) struct EventStreamServer {
 
 impl EventStreamServer {
     pub(crate) fn new(config: Config, api_version: Version) -> Result<Self, ListeningError> {
+        let address = utils::resolve_address(&config.address).map_err(|error| {
+            warn!(
+                %error,
+                address=%config.address,
+                "failed to start event stream server, cannot parse address"
+            );
+            ListeningError::ResolveAddress(error)
+        })?;
+
         let (sse_data_sender, sse_data_receiver) = mpsc::unbounded_channel();
-        let builder = utils::start_listening(&config.address)?;
+
+        // Event stream channels and filter.
+        let (broadcaster, new_subscriber_info_receiver, sse_filter) =
+            sse_server::create_channels_and_filter(config.broadcast_channel_size);
+
+        let (shutdown_sender, shutdown_receiver) = oneshot::channel::<()>();
+
+        let (address, server_with_shutdown) = warp::serve(sse_filter)
```

Comment:
In line 64 we also have an `address` member. Isn't this shadowing the previous one? Does it mean we don't need one in line 64?

Reply:
Done in [e635dcc](https://github.com/CasperLabs/casper-node/pull/1140/commits/e635dcc00edd2194dc21bf7c51bff70af392ade8).